### ISSUE TITLE
Project 3:The application of YOLO in estimating mall traffic

### ISSUE THREAD

**Issue by htyaamazing** [htyaamazing/Blog#4]
### Introduction

In the bustling world of retail, understanding the dynamics of mall traffic is not just about counting heads; it's about unlocking the heartbeat of consumer behavior. This fascination with the ebb and flow of shoppers led us to embark on a project titled "The Application of YOLO in Estimating Mall Traffic". The retail industry thrives on the footfall of customers, and accurately estimating this movement can significantly enhance operational efficiency, marketing strategies, and overall customer experience. My choice to utilize YOLO (You Only Look Once) for this task was motivated by its remarkable speed and accuracy in object detection tasks, making it an ideal candidate for real-time people counting in complex environments like malls.

At the same time, I chose to use a dataset called Crowd Counting, considering this dataset can be helpful to my project. The dataset is composed by RGB images of frames in a video (as inputs) and the object counting on every frame,  this is the number of pedestrians (object) in the image. The images are 480x640 pixels at 3 channels of the same spot  recorded by a webcam in a mall.

Here is an example of the images:

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/0061ed5e-b1da-49ed-a82a-f60e069156d1" >
</p>

### Approach

In my quest to estimate mall traffic through computer vision, I adopted a selective data utilization strategy due to the expansive size of the dataset at hand. Recognizing the challenge of processing the entirety of the data, I chose to extract representative images at half-hour intervals. This approach was designed to effectively capture the varying dynamics of mall traffic, ensuring both peak and off-peak hours were adequately represented.

To integrate computer vision into my analysis, I utilized YOLO (You Only Look Once), a cutting-edge object detection system renowned for its efficiency and accuracy. I adjusted YOLO's configuration to not only detect individuals within the mall environment but also to modify the source code so that the number of detected persons would be directly annotated on the output images. This modification was pivotal, as it provided immediate feedback on the population count, enhancing the interpretability of the results.

In refining my methodology, I diligently recorded the detected number of individuals along with the corresponding timestamps into an Excel spreadsheet. This careful organization of data streamlined the analysis process and facilitated a detailed temporal analysis of mall traffic. By correlating the people count with specific time intervals, I was able to identify patterns, determine peak traffic times, and assess the impact of external factors on the flow of mall visitors.

Here is an example of detected pictures:

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/4eab6aec-4ed7-4bf9-b058-5a64d92bafc9" >
</p>

As we can see, the people in the input pictures were well identified and recorded.

### Result & Analysis

Based on the above results, I got an Excel table, and by drawing the chart, I got the following results:

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/f017895b-0012-4cef-8839-823cf05af2da" >
</p>

The graph shows fluctuations in the headcount at different times of the day, with the number of people varying from as low as 11 to as high as 34. It is evident that there are periods with higher foot traffic, particularly around 19:00, where the headcount peaks at 34. This suggests that this time may coincide with after-work shopping or leisure activities. Early hours such as 09:00 and late hours after 21:00 show the least amount of traffic, which aligns with typical mall opening and closing times.

**Analysis:**
The data reveals several key insights about consumer behavior and mall operations:

1. Peak Hours: The highest foot traffic in the mall occurs in the evening, around 19:00, which could be attributed to people visiting the mall after work or school.

2. Off-Peak Hours: The periods between 09:00-11:00 and after 21:00 are the least busy times, which could be leveraged by mall management for activities that are best done with fewer people around, like maintenance or restocking.

3. Lunchtime Activity: There is a noticeable increase around 13:00, which could indicate a lunchtime rush, as people might be visiting food courts or shopping during their break.

4. Afternoon Consistency: The afternoon hours from 15:00 to 18:00 show a consistent headcount, suggesting a steady flow of visitors, possibly due to a mix of different demographics, including students, professionals, and families.

5. Evening Decline: After the peak at 19:00, there is a gradual decline in numbers, which may suggest the closing hours of stores or the end of happy hour promotions.

### Follow-up thinking

In the implementation of the YOLO algorithm for estimating mall traffic, the experiment's results demanded a nuanced understanding of the model's capabilities and limitations. Here's an analysis that combines both the algorithmic adjustments made during the project and the insights gleaned from the experimental outcomes:

**Model Parameter Tuning:** To enhance the accuracy of person detection, I adjusted the model parameters, particularly the confidence threshold. A higher threshold helped reduce false positives, ensuring that only individuals with a high probability were counted. This was essential to maintain the integrity of the traffic count, as evidenced by the consistent detection rates during peak hours.

**Insights from Algorithm Performance:** The experiment revealed that while YOLO is a powerful tool for object detection, its performance is highly dependent on the quality of the input data and the specific configuration of the model. The need for careful calibration of the model parameters became evident, highlighting the delicate balance between detection speed and accuracy, which is especially critical in a real-time application setting.

In conclusion, the application of the YOLO algorithm in this context taught me the importance of adapting computer vision algorithms to the specifics of the task at hand. It emphasized the necessity for meticulous preprocessing, strategic parameter tuning, and model robustness, all of which were critical in obtaining reliable results. These insights are invaluable for future projects that aim to deploy computer vision algorithms in real-world environments.
